subsys/dfu/img_util/flash_img.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "fota/flash_block"
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_ERROR
+#include <logging/sys_log.h>
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <flash.h>
+#include "boot/mcuboot_constraints.h"
+#include "img_util/flash_block.h"
+
+BUILD_ASSERT_MSG((CONFIG_IMG_BLOCK_BUF_SIZE % FLASH_ALIGN == 0),
+		 "CONFIG_IMG_BLOCK_BUF_SIZE is not multiple of FLASH_ALIGN");
+
+static bool flash_verify(struct device *dev, off_t offset,
+			 u8_t *data, size_t len)
+{
+	size_t size;
+	u32_t temp;
+	int rc;
+
+	while (len) {
+		size = (len >= sizeof(temp)) ? sizeof(temp) : len;
+		rc = flash_read(dev, offset, &temp, size);
+		if (rc) {
+			SYS_LOG_ERR("flash_read error %d offset=0x%08x",
+				    rc, offset);
+			break;
+		}
+
+		if (memcmp(data, &temp, size)) {
+			SYS_LOG_ERR("offset=0x%08x VERIFY FAIL\n\r"
```

Comment:
Awesome, thanks for adding expected/actual.

However, adding the \n\r will confuse sys log parsers (including humans), which expect each log message to be on one line (and may or may not use \n -- see SYS_LOG_NO_NEWLINE). Can you delete the \n\r?